diff --git a/gsl-config.bat b/gsl-config.bat
index d4950d0..5628740 100644
--- a/gsl-config.bat
+++ b/gsl-config.bat
@@ -2,9 +2,9 @@
 rem simple Windows version of gsl-config for conda installs
 set ROOT=%CONDA_PREFIX%\Library

-set XCFLAGS=-I"%ROOT%\include"
-set XLIBS1=-L"%ROOT%\lib" -lgsl -lgslcblas
-set XLIBS2=-L"%ROOT%\lib" -lgsl
+set XCFLAGS=-I%ROOT%\include
+set XLIBS1=-L%ROOT%\lib -lgsl -lgslcblas
+set XLIBS2=-L%ROOT%\lib -lgsl
 set XVERSION=2.3
 set XPREFIX=%ROOT%

diff --git a/setup.py b/setup.py
index 5e3946f..fb08687 100644
--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,6 @@ $ python setup.py sdist

 import distutils
 import os
-import platform
 import re
 import subprocess
 from pathlib import Path
@@ -22,6 +21,9 @@ import numpy
 from Cython.Build import cythonize

 ROOT = Path(__file__).parent
+WINDOWS = os.name == "nt"
+
+os.environ['PATH'] = "{}:{}".format(os.getenv("PATH", ""), str(ROOT.absolute()))


 def readfile(filename):
@@ -32,10 +34,12 @@ def readfile(filename):
 def gsl_config(*args, **kwargs):
     """Run gsl-config and return pre-formatted output
     """
-    return subprocess.check_output(
-        ["gsl-config"] + list(args),
-        **kwargs,
-    ).decode("utf-8").strip()
+    if WINDOWS:
+        cmd = "gsl-config {}".format(" ".join(args))
+        kwargs.setdefault("shell", True)
+    else:
+        cmd = ["gsl-config"] + list(args)
+    return subprocess.check_output(cmd, **kwargs).decode("utf-8").strip()


 def find_version():
@@ -57,7 +61,7 @@ extra_compile_args = [
     "-Wall",
     "-DHAVE_PYTHON_LINTEGRATE",
 ]
-if platform.system().lower() == "windows":
+if WINDOWS:
     extra_compile_args += [
         "-O2",
         "-DGSL_DLL",
